spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWriteBuilder.java

```diff
@@ -131,7 +132,19 @@ public Write build() {
         SparkUtil.TIMESTAMP_WITHOUT_TIMEZONE_ERROR);
 
     Schema writeSchema = SparkSchemaUtil.convert(table.schema(), dsSchema);
-    TypeUtil.validateWriteSchema(table.schema(), writeSchema, writeConf.checkNullability(), writeConf.checkOrdering());
+    boolean mergeSchema = writeInfo.options().getBoolean("mergeSchema", false);
```

Comment:
Is it a commonly defined option name? Or should it be `merge-schema`?

Reply:
This is a property name used by other sources.

Reply:
I implemented both.